### PR TITLE
Add Arm64 encoding for groups IF_SVE_CV_3A and IF_SVE_CV_3B

### PR DESCRIPTION
**Pull request by SwapnilGaikwad**
This groups emit splice instructions.

This clr output matches the one from capstone.

```
...
splice z0.b, p0, { z30.b, z31.b }
splice z3.d, p7, { z27.d, z28.d }
splice z1.h, p1, z1.h, z29.h
splice z2.s, p6, z2.s, z28.s
...
```

Contribute towards #94549.

### CODE REVIEW

**Comment by SwapnilGaikwad** [comment]
@a74nh @kunalspathak @dotnet/arm64-contrib

File: src/coreclr/jit/instr.h [hunk 1]
```diff
@@ -375,6 +375,8 @@ enum insScalableOpts : unsigned
     INS_SCALABLE_OPTS_LSL_N,               // Variants with a LSL #N (eg {<Zt>.<T>}, <Pg>, [<Xn|SP>, <Xm>, LSL #2])
     INS_SCALABLE_OPTS_MOD_N,               // Variants with a <mod> #N (eg {<Zt>.S }, <Pg>, [<Xn|SP>, <Zm>.S, <mod> #2])
 
+    INS_SCALABLE_OPTS_WITH_VECTOR_PAIR,    // Variants with {<Zn1>.<T>, <Zn2>.<T>} sve register pair (eg splice)
```
Comment (amanasifkhalid) [inline src/coreclr/jit/instr.h:378]: Not needed for this PR, but if we start to rack up a lot of `insScalableOpts`, we can probably combine this one with `INS_SCALABLE_OPTS_WITH_PREDICATE_PAIR` into something like `INS_SCALABLE_OPTS_WITH_REGISTER_PAIR`.

**Review by amanasifkhalid — APPROVED** [review]
LGTM, thanks!

File: src/coreclr/jit/emitarm64.cpp [hunk 2]
```diff
@@ -14970,6 +14988,7 @@ void emitter::emitIns_Call(EmitCallType          callType,
         case INS_sve_ld2w:
         case INS_sve_ld2d:
         case INS_sve_ld2q:
+        case INS_sve_splice: // SVE_CV_3A
```
Comment (kunalspathak) [inline src/coreclr/jit/emitarm64.cpp:14991]: looking at this method, not sure why there is inconsistency regarding comments. Are we commenting only the instructions that has multiple formats?

**Review by kunalspathak — APPROVED** [review]
LGTM

**Pull request merged**